xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -104,8 +104,9 @@ public void createXdsClientAndStart() throws IOException {
         throw new XdsInitializationException("No management server provided by bootstrap");
       }
     } catch (XdsInitializationException e) {
-      reportError(Status.fromThrowable(e));
-      throw new IOException(e);
+      IOException ioException = new IOException(e);
+      reportError(ioException);
```

Comment:
We shouldn't generally be reporting the error _and_ throwing. This code and the next change are suspicious. It looks like this should _just_ throw, as it will cause server.start() to throw.